### PR TITLE
Warn user about using default base image in v3

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

* Default base image is old, using it in v3 definition might
  result in build failures. It is good idea to fail before proceeding.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
src/ansible_builder/user_definition.py
test/unit/test_user_definition.py

### CODE REVIEW

File: src/ansible_builder/user_definition.py [hunk 1]
```diff
@@ -245,6 +245,15 @@ def validate(self):
                     self.builder_image = ImageDescription(images, 'builder_image')
                     self.build_arg_defaults['EE_BUILDER_IMAGE'] = self.builder_image.name
 
+            if (
+                self.version >= 3
+                and self.build_arg_defaults["EE_BASE_IMAGE"] == constants.build_arg_defaults["EE_BASE_IMAGE"]
+            ):
+                logging.warning(
+                    "Using the outdated base image '%s' might "
+                    "result in the build failures.", self.build_arg_defaults['EE_BASE_IMAGE']
```
Comment (Shrews) [inline src/ansible_builder/user_definition.py:254]:
```suggestion
                    "result in build failures.", self.build_arg_defaults['EE_BASE_IMAGE']
```

Comment (Shrews) [reply src/ansible_builder/user_definition.py:254]: Downstream AAP packaged builder patches the value for `EE_BASE_IMAGE` in `constants.py`. We likely don't want this warning appearing in that case. Maybe we should just hardcode the quay image string here?

**Review by Shrews — CHANGES_REQUESTED** [review]
Oops, sorry I missed reviewing this.

File: src/ansible_builder/user_definition.py (hunk 1 above)
Comment (Akasurde) [reply src/ansible_builder/user_definition.py:254]: Done.

**Review by Shrews — APPROVED** [review]

**Pull request merged**